### ISSUE TITLE
Strange Behaviour in initialPage when defaultScale params

### ISSUE THREAD

**Issue by JSurquin** [react-pdf-viewer/react-pdf-viewer#164]
Hi 
I created a codesandbox to expose the problem very quickly (sorry for the code)
when i use initialPage params with params defaultScale = {value}
everything is OK
**but** if I set up the logic to save the page viewed by the user and therefore to change the initialPage, I never come across the right page

Simple explanation: launch the codesanbox, look (for example) on page 3, when you reload the page you should fall on page 3, which is not at all the case

**If I remove the defaultScale params everything is fine**

https://codesandbox.io/s/strange-behaviour-react-pdf-viewer-initialpage-when-defaultscale-params-g8p1m?file=/src/App.js:3467-3498

**Comment by JSurquin** [comment]
currently for it to work, so I have to keep the default behavior and therefore the maximized page in its space
whereas I would like to fix on 100% (it does not work either with fit etc)

**Comment by phuoc-ng** [comment]
@JSurquin Do you mean that issue when reloading the page and scrolling?

<img width="669" alt="Screen Shot 2020-05-11 at 22 22 16" src="https://user-images.githubusercontent.com/57786711/81579090-eb1bb480-93d5-11ea-8c3b-306a02a4b2e1.png">

**Comment by JSurquin** [comment]
Hi , no
it is really very difficult to explain, in clear, it must have a problem of calculation in the width or the height of the page generated according to the resolution of the screen, I make 2 gif

---
![work](https://user-images.githubusercontent.com/33385043/82031459-35cd5300-969a-11ea-9e27-2b6f19f9de72.gif)

The first, WITH, the params defaultScale = {1.1}
when the width of the screen is reduced by half (a tablet for example, as seen on the gif I have side by side my browser and a notepad, as we can see, I fall back well on the same page

---
![dontwork](https://user-images.githubusercontent.com/33385043/82031492-3fef5180-969a-11ea-828e-94b802a894e8.gif)
The 2nd, I am in full screen (resolution 1920x1080 with zoom of the font size)
and as we can see, I come back randomly on a page

**Comment by phuoc-ng** [comment]
@JSurquin I just added `onPageChange` option that will be called when the user change the current page.

So, you can store the current page and load it as seen in this [example](https://github.com/phuoc-ng/react-pdf-viewer/blob/master/demo/store-current-page/StoreCurrentPageExample.tsx).

You can check it out and see if the problem still exists with the new approach.

**Comment by JSurquin** [comment]
Thank you i will watch from tomorrow

**Comment by aholland-carecloud** [comment]
I was running into the same issue. I created a minimum reproducible example. 
https://codesandbox.io/s/strange-behaviour-react-pdf-viewer-initialpage-and-defaultscale-props-869yt

I was playing around with it using different pdfjs worker versions that didn't seem to have any effect. What does have an effect is the width of the viewer. It will open to the incorrect initialPage when the defaultScale is also passed in.

Few things I noticed. 
- It initially for a brief second goes to the correct page but then changes (looks like changing the scale changes the current page). Might just be the order, may need to default scale needs to be set before the initial page.
- It only gets the wrong initialPage if you pass in a page that is in the middle of the range. It doesn't seem to happen on initial page 0-2 and 13-15 of this 16 page document. Not sure how much the viewer width effects this.
- The viewer width being wider causes it to be off more number of pages.
